packages/shared-metrics/src/dependencies.js

```diff
@@ -269,3 +310,83 @@ function sortByDistance(distances, dependency1, dependency2) {
   }
   return d2 - d1;
 }
+
+/**
+ * Either reads the contents of the main package.json file or returns mainPackageJsonContents if it was cached already
+ * @param {string} mainPackageJsonPath
+ * @param {(err: Error, contents: string) => void} cb
+ */
+function getMainPackageJsonContents(mainPackageJsonPath, cb) {
+  if (mainPackageJsonContents) {
+    cb(null, mainPackageJsonContents);
+  } else {
+    fs.readFile(mainPackageJsonPath, { encoding: 'utf8' }, (readFileErr, contents) => {
+      if (readFileErr) {
+        logger.debug(`Error reading ${mainPackageJsonPath}`, readFileErr);
+        cb(readFileErr, null);
+      } else {
+        mainPackageJsonContents = contents;
+        cb(null, mainPackageJsonContents);
+      }
+    });
+  }
+}
```

Comment:
_suggestion:_ We already have facilities to read and JSON-parse the main package.json files in applicationUnderMonitoring.js: https://github.com/instana/nodejs/blob/d8c48eaa9aa78e9fde2d680f841627ec5a583fac/packages/core/src/util/applicationUnderMonitoring.js#L69-L80

It is currently a part of the `getMainPackageJsonStartingAtDirectory` that can not be used independently. Could you extract lines 69-80 of applicationUnderMonitoring into its own exported method and use it here?

Reply:
good idea. leave it to me!